bmorph/core/workflows.py

```diff
@@ -183,260 +75,79 @@ def apply_interval_bmorph(raw_ts, train_ts, obs_ts,
     """
     assert isinstance(bmorph_interval, pd.DateOffset)
 
+    if condition_ts is not None:
+        raw_y = condition_ts
+        ref_y = condition_ts
+        train_y = condition_ts
+
     if bmorph_interval == pd.DateOffset(days=1):
         raise Exception("Please enter a bmorph_interval greater than 1 day(s)")
 
-    train_window = slice(*train_window)
-    bmorph_window = slice(*bmorph_window)
-    reference_window = slice(*reference_window)
+    raw_train_window = slice(*raw_train_window)
+    apply_window = slice(*apply_window)
+    ref_train_window = slice(*ref_train_window)
     raw_ts_window = slice(pd.to_datetime(raw_ts.index.values[0]),
                           pd.to_datetime(raw_ts.index.values[-1]))
 
     # bmorph the series
     overlap_period = int(bmorph_overlap / 2)
     bmorph_ts = pd.Series([])
     bmorph_multipliers = pd.Series([])
-    bmorph_range = pd.date_range(bmorph_window.start, bmorph_window.stop+bmorph_interval,
+    bmorph_range = pd.date_range(apply_window.start,
+                                 apply_window.stop+bmorph_interval,
                                  freq=bmorph_interval)
     for i in range(0,len(bmorph_range)-1):
         bmorph_start = bmorph_range[i]
         bmorph_end = bmorph_range[i+1]
 
-        # we don't need ot adjust for overlap if it is the last entry at i+1
-        if i < len(bmorph_range)-2:
-            bmorph_end -= pd.DateOffset(days=1)
+        ## we don't need to adjust for overlap if it is the last entry at i+1
```

Comment:
if this is no longer being used it should be deleted_

Reply:
same for 104 and 105

Reply:
Done